### DISCUSSION TITLE
Bootloader for robin nano v2.2

### DISCUSSION THREAD

**Discussion by PolitehPetra** [NARUTOfzr/Neptune_3#81] [Q&A]
After replacing the microcontroller on the Robin Nano v2.2 board, I can't install the firmware. I'm trying to load the bootloader for robin nano 2.x, but it gives an error that there is not enough memory

**Comment by PolitehPetra** [comment]
Even manufacturers don't want to share the bootloader for their board with me. Can anyone tell me how to get around this? I have a programmer. I understand that each bootloader has its own rules for the firmware file, its location and name. I initially had the Klipper firmware and I need to somehow put it into the board, I can't do it directly from the programmer, since there is some special format from the Klipper itself. What bootloader can I install and how to name the firmware file for the STM32F401RCT6 microcontroller?

**Comment by Saka05** [comment]
i won't choose Elegoo for this reason alone!
